server/user-bootstrap/index.js

```diff
@@ -27,53 +27,56 @@ const debug = debugFactory( 'calypso:bootstrap' ),
 	},
 	url = `${ API_PATH }?${ stringify( apiQuery ) }`;
 
-module.exports = function( authCookieValue, geoCountry, callback ) {
-	// create HTTP Request object
-	const req = superagent.get( url );
-	let hmac, hash;
+module.exports = function( authCookieValue, geoCountry ) {
+	return new Promise( ( resolve, reject ) => {
+		// create HTTP Request object
+		const req = superagent.get( url );
+		let hmac, hash;
 
-	if ( authCookieValue ) {
-		authCookieValue = decodeURIComponent( authCookieValue );
+		if ( authCookieValue ) {
+			authCookieValue = decodeURIComponent( authCookieValue );
 
-		if ( typeof API_KEY !== 'string' ) {
-			callback( new Error( 'Unable to boostrap user because of invalid API key in secrets.json' ) );
-			return;
-		}
+			if ( typeof API_KEY !== 'string' ) {
+				reject( new Error( 'Unable to boostrap user because of invalid API key in secrets.json' ) );
```

Comment:
You can probably move the `return` from the line below here:

```js
return reject( new Error( 'Unable to boostrap user because of invalid API key in secrets.json' ) );
```